internal/store/fleet.go

```diff
@@ -122,9 +122,21 @@ func (s *FleetStore) ListIgnoreOrg() ([]model.Fleet, error) {
 }
 
 func (s *FleetStore) DeleteAll(ctx context.Context, orgId uuid.UUID) error {
-	condition := model.Fleet{}
-	result := s.db.Unscoped().Where("org_id = ?", orgId).Delete(&condition)
-	return result.Error
+	err := s.db.Transaction(func(tx *gorm.DB) error {
+		condition := model.Fleet{}
+		result := s.db.Unscoped().Where("org_id = ?", orgId).Delete(&condition)
+		if result.Error != nil {
+			return result.Error
+		}
+
+		// Unset owner for all devices in org, now that all fleets have been deleted
+		devCondition := model.Device{
+			Resource: model.Resource{OrgID: orgId},
+		}
+		result = tx.Model(devCondition).Select("owner").Updates(map[string]interface{}{"owner": nil})
+		return result.Error
```

Comment:
I think we should make this a call to the Device store instead, the generation should be incremented, and probably the change also tracked in the status.condition updates.

Reply:
I can make it call the device store.  We need to discuss device generations and conditions, as per the comment above.

Reply:
@avishayt You should add `.Where()` otherwise you will remove owner from all.
`tx.Model(devCondition).Where(devCondition).Select("owner").Updates(map[string]interface{}{"owner": nil})`

Reply:
Code was removed